micrometer-core/src/main/java/io/micrometer/core/instrument/binder/neo4j/Neo4jMetrics.java

```diff
@@ -0,0 +1,111 @@
+        this.driver = driver;
+        this.tags = Tags.concat(tags, "name", name);
+    }
+
+    @Override
+    public void bindTo(MeterRegistry meterRegistry) {
+        Metrics metrics = this.driver.metrics();
+        metrics.connectionPoolMetrics().forEach(this.getPoolMetricsBinder(meterRegistry));
+    }
+
+    Consumer<ConnectionPoolMetrics> getPoolMetricsBinder(MeterRegistry meterRegistry) {
+        return (poolMetrics) -> {
+            Iterable<Tag> poolTags = Tags.concat(this.tags, "poolId", poolMetrics.id());
+
+            // acquisition attempts
+            FunctionCounter.builder(PREFIX + ".acquisition", poolMetrics, ConnectionPoolMetrics::acquired)
+                    .tags(Tags.concat(poolTags, "outcome", "success"))
+                    .baseUnit(BASE_UNIT_CONNECTIONS).description("The amount of connections that have been acquired.")
+                    .register(meterRegistry);
+
+            FunctionCounter
+                    .builder(PREFIX + ".acquisition", poolMetrics, ConnectionPoolMetrics::timedOutToAcquire)
+                    .tags(Tags.concat(poolTags, "outcome", "timeout"))
+                    .baseUnit(BASE_UNIT_CONNECTIONS)
+                    .description("The amount of failures to acquire a connection from a pool within maximum connection "
+                                    + "acquisition timeout.")
+                    .register(meterRegistry);
+
+            // creation attempts
+            FunctionCounter.builder(PREFIX + ".creation", poolMetrics, ConnectionPoolMetrics::created)
+                    .tags(Tags.concat(poolTags, "outcome", "success"))
+                    .baseUnit(BASE_UNIT_CONNECTIONS).description("The amount of connections have ever been created.")
+                    .register(meterRegistry);
+
+            FunctionCounter.builder(PREFIX + ".creation", poolMetrics, ConnectionPoolMetrics::failedToCreate)
+                    .tags(Tags.concat(poolTags, "outcome", "failure"))
+                    .baseUnit(BASE_UNIT_CONNECTIONS)
+                    .description("The amount of connections have been failed to create.").register(meterRegistry);
+
+            // active pool size
+            Gauge.builder(PREFIX + ".current", poolMetrics, ConnectionPoolMetrics::idle)
+                    .tags(Tags.concat(poolTags, "state", "idle"))
+                    .baseUnit(BASE_UNIT_CONNECTIONS).description("The amount of connections that are currently idle.")
+                    .register(meterRegistry);
+
+            Gauge.builder(PREFIX + ".current", poolMetrics, ConnectionPoolMetrics::inUse)
+                    .tags(Tags.concat(poolTags, "state", "inUse"))
+                    .baseUnit(BASE_UNIT_CONNECTIONS).description("The amount of connections that are currently in-use.")
+                    .register(meterRegistry);
+
+            FunctionCounter.builder(PREFIX + ".closed", poolMetrics, ConnectionPoolMetrics::failedToCreate)
```

Comment:
```suggestion
            FunctionCounter.builder(PREFIX + ".closed", poolMetrics, ConnectionPoolMetrics::closed)
```

Reply:
For the first one: I really don't know what would be a good fit here.
Multiple instances could be created because of another database connection (within the same cluster)
but also there are other credentials that act as another user.

For the second one: Good catch, thanks.